src/node.rs

```diff
@@ -10,26 +10,28 @@ use crate::style::*;
 
 type MeasureFunc = Box<Fn(Size<Number>) -> Result<Size<f32>>>;
 
-pub type NodeId = usize;
-pub(crate) type Storage<T> = HashMap<NodeId, T>;
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Node(usize);
+
+pub(crate) type Storage<T> = HashMap<Node, T>;
 
 struct Allocator {
-    new_id: NodeId,
-    free_ids: Vec<NodeId>,
+    new_id: usize,
```

Comment:
might still be worth having an alias for NodeId for use in these situations. What do you think?

```
pub(crate) type NodeId = usize;
pub struct Node(NodeId);
```

Reply:
Looks good, I will change it.